src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+            }
+
+            throw new UnsupportedTemporalTypeException("Unsupported unit: " + unit);
+        }
+
+        return unit.between(this, end);
+    }
+
+    /**
+     * Get the number of years from this date to the given day.
+     *
+     * @param end The end date.
+     * @return The number of years from this date to the given day.
+     */
+    long yearsUntil(final InternationalFixedDate end) {
+        long startYear = getProlepticYear() * 512L + getDayOfYear();
+        long endYear = end.getProlepticYear() * 512L + end.getDayOfYear();
+
+        return (endYear - startYear) / 512L;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param endDateExclusive
+     * @return period up to that date
+     */
+    @Override
+    public ChronoPeriod until(final ChronoLocalDate endDateExclusive) {
+        InternationalFixedDate end = InternationalFixedDate.from(endDateExclusive);
+        int years = Math.toIntExact(yearsUntil(end));
+        // Get to the same "whole" year.
+        InternationalFixedDate sameYearEnd = end.plusYears(years);
+        int months = (int) monthsUntil(sameYearEnd);
+        int days = (int) daysUntil(sameYearEnd.plusMonths(months));
+
+        return getChronology().period(years, months, days);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param end
+     * @return period up to that date
+     */
+    @Override
+    long monthsUntil(final AbstractDate end) {
+        InternationalFixedDate date = InternationalFixedDate.from(end);
+        long monthStart = this.getCalculatedProlepticMonth() * 256L + this.getCalculatedDayOfMonth();
+        long monthEnd = date.getCalculatedProlepticMonth() * 256L + date.getCalculatedDayOfMonth();
+
+        return (monthEnd - monthStart) / 256L;
+    }
+
+    //-----------------------------------------------------------------------
+    @Override
+    public long toEpochDay() {
+        long epochDay =
+                ((long) getProlepticYear()) * InternationalFixedChronology.DAYS_IN_YEAR +
+                getLeapYearsBefore(getProlepticYear()) + getDayOfYear();
+
+        return epochDay - InternationalFixedChronology.DAYS_0000_TO_1970;
+    }
+
+    /**
+     * Display the date in human-readable format.
+     * Note: leap-day and year-day are not part of any month, thus they appear as "2008-leap-day" / "2004-year-day".
+     *
+     * TODO: the string "leap-day" and "year-day" depends on a {@link Locale}, should be loaded from a resource bundle.
+     * TODO: Some languages may prefer the wording "leap-day of 2008" rather than "2008-leap-day".
+     *
+     * @return The number of years from this date to the given day.
+     */
+    @Override
+    public String toString() {
+        StringBuilder buf = new StringBuilder(30);
+        buf.append(getChronology().toString())
+                .append(' ')
+                .append(getEra())
+                .append(' ')
+                .append(getYearOfEra());
+
+        if (isLeapDay()) {
```

Comment:
I think I prefer spaces before `"leap-day"` and `"year-day"`, but not too important.  Wikipedia has all words capitalized (and no hyphen before `"Day"`, either), so you may want to check some of the original references.